src/provider/model/database/Database.ts

```diff
@@ -0,0 +1,69 @@
+import {injectable} from 'inversify';
+import Dexie from 'dexie';
+
+import {StoredSettings} from '../StoredSettings';
+import {StoredNotification} from '../StoredNotification';
+import {DeferredPromise} from '../../common/DeferredPromise';
+import {AsyncInit} from '../../controller/AsyncInit';
+
+import {Collection} from './Collection';
+
+export const enum CollectionMap {
+    NOTIFICATIONS = 'notifications',
+    SETTINGS = 'settings'
+}
+
+export type Collections = {
+    [CollectionMap.NOTIFICATIONS]: StoredNotification;
+    [CollectionMap.SETTINGS]: StoredSettings;
+};
+
+@injectable()
+export class Database extends AsyncInit {
+    private _database: Dexie;
+    private _collections: Map<CollectionMap, Collection<any>>;
+    private _databaseReadyPromise: DeferredPromise<void>;
+
+    constructor () {
+        super();
+
+        this._database = new Dexie('Notifications_Service');
```

Comment:
'notifications-service`, to be roughly consistent with jenkins config/channel/window/ name? Or is this a Dexie convention

Reply:
Done! The other DB we tried out forced this convention and never changed it.  Ive updated to `notifications-service`